query/common/src/main/java/org/codice/ddf/admin/common/services/ServiceCommons.java

```diff
@@ -13,25 +13,134 @@
+import org.codice.ddf.admin.common.fields.common.PidField;
+import org.codice.ddf.admin.configurator.Configurator;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.admin.configurator.OperationReport;
 
 public class ServiceCommons {
 
     public static final String SERVICE_PID_KEY = "service.pid";
 
     public static final String FACTORY_PID_KEY = "service.factoryPid";
 
+    // A flag to indicate if a service being updated has a password of "password". If so, the
+    // password will not be updated.
+    public static final String FLAG_PASSWORD = "password";
+
     public String resolveProperty(String str) {
         return StrSubstitutor.replaceSystemProperties(str);
     }
 
     public List<String> resolveProperties(String... list) {
         return Arrays.stream(list)
-                .map(str -> resolveProperty(str))
+                .map(this::resolveProperty)
                 .collect(Collectors.toList());
     }
+
+    public static Report createManagedService(Map<String, Object> serviceProps, String factoryPid,
+            ConfiguratorFactory configuratorFactory) {
+        Report report = new Report();
+        Configurator configurator = configuratorFactory.getConfigurator();
+
+        configurator.createManagedService(factoryPid, serviceProps);
+        if(configurator.commit("Service saved with details [{}]", serviceProps.toString())
+                .containsFailedResults()) {
+            report.argumentMessage(failedPersistError());
+        }
+        return report;
+    }
+
+    public static Report updateService(PidField servicePid, Map<String, Object> newConfig,
+            ConfiguratorFactory configuratorFactory) {
+        Report report = new Report();
+        report.addMessages(serviceConfigurationExists(servicePid, configuratorFactory));
+        if (report.containsErrorMsgs()) {
+            return report;
+        }
+
+        String pid = servicePid.getValue();
+        Configurator configurator = configuratorFactory.getConfigurator();
+        configurator.updateConfigFile(pid, newConfig, true);
+        OperationReport operationReport = configurator.commit(
+                "Updated config with pid [{}] and new service properties [{}]",
+                pid,
+                newConfig.toString());
+        if (operationReport.containsFailedResults()) {
+            return report.argumentMessage(failedUpdateError(servicePid.path()));
+        }
+
+        return report;
+    }
+
+    public static Report deleteService(StringField servicePid, ConfiguratorFactory configuratorFactory) {
+        Report report = new Report();
+        Configurator configurator = configuratorFactory.getConfigurator();
+        configurator.deleteManagedService(servicePid.getValue());
+        if(configurator.commit("Deleted source with pid [{}].", servicePid.getValue())
+                .containsFailedResults()) {
+            report.argumentMessage(failedDeleteError(servicePid.path()));
```

Comment:
The Configurator returns "Internal Errors" and it does not use our API, so there is not much we can do about it. It can happen for many reasons, such as an IOException or not have proper privileges to do the action.